File: SecondYear/Junior-Hawk_ArchitAgarwal_2226cseai1186_2/Week_2/count_digits.cpp
int countDigits(int n){
	int k = n;
	int c=0;
	while(k>0){
		int rem = k%10;
		if(rem!=0 && n%rem==0){
			c++;
		}
		k/=10;
	}	
	return c;
}

File: SecondYear/Divyansh670_DivyanshSrivastav_2226it1029_2/Week_2/count_digits.java
import java.util.Scanner;

public class count_digits {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        scanner.close();

        int originalN = n; 
        int count = 0; 

        while (n > 0) {
            int digit = n % 10; 
            if (digit != 0 && originalN % digit == 0) {
                count++; 
            }
            n /= 10; 
        }

        System.out.println(count);
    }
}
